powertools-idempotency/src/test/java/software/amazon/lambda/powertools/idempotency/internal/IdempotencyAspectTest.java

```diff
@@ -133,14 +142,43 @@ public void secondCall_inProgress_shouldThrowIdempotencyAlreadyInProgressExcepti
                 DataRecord.Status.INPROGRESS,
                 Instant.now().plus(356, SECONDS).getEpochSecond(),
                 JsonConfig.get().getObjectMapper().writer().writeValueAsString(b),
-                null);
+                null,
+                OptionalLong.of(Instant.now().toEpochMilli() + 1000));
         doReturn(record).when(store).getRecord(any(), any());
 
         // THEN
         IdempotencyEnabledFunction function = new IdempotencyEnabledFunction();
         assertThatThrownBy(() -> function.handleRequest(p, context)).isInstanceOf(IdempotencyAlreadyInProgressException.class);
     }
 
+    @Test
+    public void secondCall_inProgress_lambdaTimeout_timeoutExpired_shouldThrowInconsistentState() throws JsonProcessingException {
+        // GIVEN
+        Idempotency.config()
+                .withPersistenceStore(store)
+                .withConfig(IdempotencyConfig.builder()
+                        .withEventKeyJMESPath("id")
+                        .build()
+                ).configure();
+
+        doThrow(IdempotencyItemAlreadyExistsException.class).when(store).saveInProgress(any(), any(), any());
+
+        Product p = new Product(42, "fake product", 12);
+        Basket b = new Basket(p);
+        DataRecord record = new DataRecord(
+                "42",
+                DataRecord.Status.INPROGRESS,
+                Instant.now().plus(356, SECONDS).getEpochSecond(),
+                JsonConfig.get().getObjectMapper().writer().writeValueAsString(b),
+                null,
+                OptionalLong.of(Instant.now().toEpochMilli() - 100));
```

Comment:
These tests are great, but take a bit of time to understand as someone new to the codebase.

Can you move `Instant.now().toEpochMilli() - 100` to a variable with a name that shows it's in the past.

Reply:
Yes, I agree it's not that easy to jump in. I tried to have explicit test names but stil...
Added a variable and a comment.